### ISSUE TITLE
MCP Inspector displays unnecessary error output even when FastMCP connects successfully

### ISSUE THREAD

**Issue by harshpreet931** [modelcontextprotocol/inspector#732]
### Description
When starting the STDIO server with FastMCP, MCP Inspector displays an error in the terminal even though the connection is established successfully and tools work as expected.

This appears to be misleading, since the error output suggests a failure, but the server is functioning correctly.

### Observed Behavior
- STDIO server starts normally
- MCP Inspector shows an error message in the terminal (screenshot below)
- Despite the error, FastMCP connects and tools are callable

<img width="640" height="1990" alt="Image" src="https://github.com/user-attachments/assets/50ee3424-defe-4cf5-bf12-c9aae5cd172e" />

### Example Code
```python
from fastmcp import FastMCP

mcp = FastMCP(name="StdioExampleServer")

@mcp.tool
def echo(message: str) -> str:
    """Echoes the provided message back to the client."""
    return f"You said: {message}"

@mcp.tool
def api_fetch(url: str) -> str: 
    """Fetches data from the provided URL."""
    import requests
    response = requests.get(url)
    return response.text if response.status_code == 200 else f"Error fetching {url}: {response.status_code}"

if __name__ == "__main__":
    mcp.run()
```

### Version Information
```
MCP Inspector version:                                                                       0.16.5
FastMCP version:                                                                                 2.11.3
MCP version:                                                                                        1.13.0
Python version:                                                                                    3.12.11
Platform:                                                                     macOS-15.6-x86_64-i386-64bit
FastMCP root path: /Users/harshpreet.singh/Desktop/Projects/MCP_Examples/venv/lib/python3.12/site-packages

```

**Comment by cliffhall** [comment]
> Having a dedicated "server logs" section in MCP Inspector with elevated visibility for errors (non-zero exit codes) would be the ideal solution. This would preserve the standard stderr logging behavior while making the UI more intuitive for users.

I'll work on this. I have some other [related changes](https://github.com/modelcontextprotocol/servers/pull/2469#issuecomment-3151317777) to make anyway.

**Comment by strawgate** [comment]
I actually meant that when the process exits with a non zero exit code that could be the time to present the log information in a more "in your face" kind of way. 

I.e.. The logs exist in a logs tab until the process exits with a non zero exit code and when that happens the location below the connect button shows the most recent log messages or perhaps a link "the process exited unexpectedly. See recent logs..."

**Comment by cliffhall** [comment]
Okay, folks. Here's [the PR](https://github.com/modelcontextprotocol/inspector/pull/741). 

Turns out, there's not going to be a separate tab / panel for `stderr` output. The way those messages were being differentiated from normal logging notifications on the client was that there was an out-of-spec message type being used to send them (`notifications/stderr` instead of `notifications/message`). My  The only way to communicate with the client is via JSON-RPC messages, and the valid type for logging is what I'm sending. I am, however, inspecting the message, so if it includes an RFC 5424 Syslog Protocol level, it will be sent with that level and therefore be filterable by the log level filter setting in the client.
